### ISSUE TITLE
Accessibility failure: instantSearch Panel widget button has no text for screen readers

### ISSUE THREAD

**Issue by rachel-simpleclick** [algolia/instantsearch.js#6663]
### 🐛 Current behavior

Screen reader users are unable to see the context of the panel widget's button.

As far as I can see there is no ability to add text or an aria-label to the button template.

I have been asked by a client to make their site compliant with the European Accessibility Act (EAA) that came into affect 28 June 2025, so need some way to get some context onto the button.

https://accessible-eu-centre.ec.europa.eu/content-corner/news/eaa-comes-effect-june-2025-are-you-ready-2025-01-31_en

If using chrome, using the ARIA DevTools plugin, illustrates nicely what a screenreader can see and shows the button with no context.

### 🔍 Steps to reproduce

1 - Add an instantSearch panel wiget to page.
2 - Render
3 - Inspect the html - the <button class="ais-Panel-collapseButton"> has no text within it, only the icon

### Live reproduction

https://codesandbox.io/p/sandbox/cool-volhard-3ylhvc

### 💭 Expected behavior

Ability to add in any of the follwing via options on the collapseButtonText template:
- button text
- button aria-label
- button id (to allow the use of aria-labelled-by e.g. on the header template)

### Package version

5.1.1

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
